app/Repositories/FileBaseRepository.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace App\Repositories;
+
+class FileBaseRepository
+{
+    protected $dirName;
+
+    public function __construct($fileName, $dirName)
+    {
+        $this->fileName = $fileName;
+
+        if (!is_dir($dirName)) {
+            mkdir($dirName);
+        }
+
+        fopen(prepare_file_path("{$dirName}/{$this->fileName}"), 'a+');
```

Comment:
let's create the $filePath class field to use it in all child classes

```suggestion
        $this->filePath = prepare_file_path("{$dirName}/{$this->fileName}");

        fopen($this->filePath, 'a+');
```